exercises/run-length-encoding/src/example/java/RunLengthEncoding.java

```diff
@@ -0,0 +1,65 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RunLengthEncoding {
+    public String encode(String data) {
+
+        StringBuilder encodedData = new StringBuilder();
+
+        if (data.length() == 0) {
+            return data;
+        }
+
+        int charCount = 0;
+        char beforeChar = data.charAt(0);
+
+        for (int i = 0; i < data.length(); i++) {
+            char currentChar = data.charAt(i);
+

```

Comment:
I think there are too many ifs here. From what I understand, you only really need to do two things; if `beforeChar == currentChar` then you need to increment the count. If `beforeChar != currentChar` or this is the last char in the string, then we need to add the latest char with count to the string. So something like (in pseudocode):
```
if (beforeChar == currentChar) {
  increment count
}

if (beforeChar != currentChar || reached end of string) {
  add beforeChar with count
  if beforeChar != currentChar && reached end of string then add currentChar with count 1
  beforeChar = currentChar
  charCount = 1
}
```

Extracting the adding char with count to its own method would also simplify it :)

What do you think? :)